kb/2022-03-10/multiple_nics_vm_connectivity.md

```diff
@@ -0,0 +1,148 @@
+
+Once connected, you can run again the same command as before:
+```bash
+ip route show default
+```
+
+However, this time you'll get a default route with an incorrect gateway IP.
+For example:
+```
+default via <Incorrect Gateway IP> dev eth0 proto dhcp src <VM's IP> metric 100
+```
+
+## Why does connectivity issues occurs randomly
+
+In a standard setup, cloud based VMs are very often configured with DHCP for all NICs by default. The configuration sets an IP and a gateway for each NIC. Lastly, a default route to the gateway IP will also be added, so you can connect to the VM using its IP.
+
+However, Linux distributions start multiple DHCP clients simultaneously and don't have a "prioritization" system. This means that in the case you have two or more NICs configured with DHCP, the clients will enter a "race condition" to configure the default route. And depending on the Linux distribution DHCP script currently running, there's no guarantee on which default route will be configured.
+
+As the default route might change in every DHCP renewing process or after every OS reboot, this will create the network connectivity issues.
+
+## How to avoid the random connectivity issues
+
+You could avoid these connectivity issues easily by having only one NIC attached to the VM and have only one IP and one gateway configured.
+
+However, the use of only one NIC is often not possible for VMs in a more complex infrastructure. For example, if your infrastructure has a storage network and a service network. For security reasons, the storage network will be isolated from the service network, and have an independent subnet. In this situation, you must have two NICs to connect to both the service and storage networks.
+
+As a solution, you can choose one solution from below to fit your requirement and security policy.
+
+### Disable DHCP on secondary NIC
+
+As explained above, the issue is caused by a "race condition" of two DHCP clients. One solution to avoid this issue, is to disable DHCP for the all NICs and configure them only with a static IP. In the same way, you could configure the secondary NIC with a static IP and keep the primary NIC with DHCP enabled.
+
+1. To configure the primary NIC with a static IP (`eth0` in this example), you can edit the file `/etc/sysconfig/network/ifcfg-eth0` with the following values:
+
+```
+BOOTPROTO='static'
+IPADDR='192.168.0.100'
+NETMASK='255.255.255.0'
+```
+
+Or, if you want to keep the primary NIC with DHCP (`eth0` in this example), use the following values instead:
+
+```
+BOOTPROTO='dhcp'
+DHCLIENT_SET_DEFAULT_ROUTE='yes'
+```
+
+
+2. You'll need to configure the default route by editing the file `/etc/sysconfig/network/ifroute-eth0` (skip this step if you configured your primary NIC with DHCP):
+
+
+```
+# Destination  Dummy/Gateway  Netmask  Interface
+default        192.168.0.254  -        eth0
+```
+
+!!! warning
+    Do not put other default route for your secondary NIC
```

Comment:
ditto

Reply:
I will keep this as warning type.

Reply:
sorry I mean the syntax here, if u run it with `yarn install && yarn start` u will have a preview of the blog, thanks.

Reply:
got it